optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/bi/BavetGroupBiConstraintStream.java

```diff
@@ -50,19 +50,20 @@ public BavetGroupBiConstraintStream(BavetConstraintFactory<Solution_> constraint
 
     @Override
     public BavetGroupBiNode<GroupKey_, ResultContainer_, Result_> createNodeChain(BavetNodeBuildPolicy<Solution_> buildPolicy,
-            Score<?> constraintWeight, int nodeOrder, BavetAbstractBiNode<GroupKey_, Result_> parentNode) {
+            Score<?> constraintWeight, BavetAbstractBiNode<GroupKey_, Result_> parentNode) {
         return (BavetGroupBiNode<GroupKey_, ResultContainer_, Result_>) super.createNodeChain(buildPolicy, constraintWeight,
-                nodeOrder, parentNode);
+                parentNode);
     }
 
     @Override
     protected BavetGroupBiNode<GroupKey_, ResultContainer_, Result_> createNode(BavetNodeBuildPolicy<Solution_> buildPolicy,
-            Score<?> constraintWeight, int nodeOrder, BavetAbstractBiNode<GroupKey_, Result_> parentNode) {
+            Score<?> constraintWeight, BavetAbstractBiNode<GroupKey_, Result_> parentNode) {
         if (parentNode != null) {
             throw new IllegalStateException("Impossible state: the stream (" + this
                     + ") cannot have a parentNode (" + parentNode + ").");
         }
-        return new BavetGroupBiNode<>(buildPolicy.getSession(), nodeOrder, finisher);
+        // GroupBridgeNodes are created after GroupNodes, so increment by 2 to create room for the bridge inbetween.
+        return new BavetGroupBiNode<>(buildPolicy.getSession(), buildPolicy.getNodeIndexMaximum() + 2, finisher);
```

Comment:
Hell no. This appraoch won't work with composition approaches.

Reply:
Assuming that Group is always preceded by a GroupBridge, this is perfectly safe.
Otherwise I'll need to see an example of a network where that is not the case.